gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/KubernetesExtensionPropertyTest.java

```diff
@@ -171,21 +47,182 @@ public void setUp() throws Exception {
         .build();
   }
 
-  @Test
-  public void getValue_withDefaults_shouldReturnDefaultValue() throws Exception {
+  @ParameterizedTest(name = "{index}: {0} with defaults returns ''{1}''")
+  @MethodSource("defaultValues")
+  void getValue_withDefaults_shouldReturnDefaultValue(String method, Object expectedDefault) throws Exception {
     // When
     final Object result = extension.getClass().getMethod(method).invoke(extension);
     // Then
     assertThat(result).isEqualTo(expectedDefault);
   }
 
-  @Test
-  public void getValue_withProperty_shouldReturnFromPropertyValue() throws Exception {
+  static Stream<Arguments> defaultValues() {
+    return Stream.of(
```

Comment:
Some arguments appear to be missing here, initially there were 70 parameters.
This method returns only 69.
Any explanation for this?

Reply:
OK; I see it now, there's a duplicate entry for `getBuildStrategyOrDefault` that only applies for the properties set `jib`, `s2i`.

> Since we've another test that will check for build strategy set by the property.

However, you'd probably want to add the third option to the `propertiesAndValues` set:
```java
        arguments("getBuildStrategyOrDefault", "jkube.build.strategy", "docker", JKubeBuildStrategy.docker),
```

The reasoning is that you need to consider the following spec:
- The default build strategy if not specified is `docker`
- I can set a build strategy by providing the property `jkube.build.strategy`
  - with value `s2i` will use S2I
  - with value `jib` will use Jib
  - with value `docker` will use Docker _(missing)_

Considering a future change in the spec (new default, additional strategy, removed strategy, and so on), I could easily change the tests to match the updated spec.

Reply:
Added 👍